### PR TITLE
Extensions README.md for non-Go devs

### PR DESCRIPTION
**Pull request by TristonianJones**
This file was trivially generated with `go doc github.com/google/cel-go/ext.Strings` and has a few hand edits to convert the format to an appropriate `README.md`

### CODE REVIEW

**Comment by TristonianJones** [comment]
@bigkevmcd FYI

**Comment by bigkevmcd** [comment]
@TristonianJones thank you! both for this, and your efforts in adding the string functions.

Looks good to me.

File: ext/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    'hello mellow'.indexOf('ello', 2)  // returns 7
+    'hello mellow'.indexOf('ello', 20) // error
+
+### LastIndexOf
+
+Returns the integer index of the last occurrence of the search string. If the
+search string is not found the function returns -1.
+
+The function also accepts an optional index which represents where to begin the
+substring search starting from the end of the string. If the substring is the
```
Comment (JimLarson) [inline ext/README.md]: See final comments on #319.  Perhaps mention that for all strings `S` and `T` we should have:

```
S.indexOf(T, P) + len(T) + rev(S).lastIndexOf(rev(T), len(S) - len(T) - P) = len(S)
```

when `S.indexOf(T, P)` is nonnegative.

**Review by JimLarson — CHANGES_REQUESTED** [review]

File: ext/README.md (hunk 1 above)
Comment (TristonianJones) [reply ext/README.md]: PTAL

File: ext/strings_test.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ var stringTests = []struct {
 	{expr: `'hello wello'.lastIndexOf('elbo room!!') == -1`},
 	{expr: `'hello wello'.lastIndexOf('elbo room!!!') == -1`},
 	{expr: `'hello wello'.lastIndexOf('hello wello') == 0`},
+	{expr: `'bananananana'.lastIndexOf('nana', 8) == 8`},
```
Comment (JimLarson) [inline ext/strings_test.go]: Change it to `'bananananana'.lastIndexOf('nana', 7) == 6`

**Review by JimLarson — APPROVED** [review]

File: ext/strings_test.go (hunk 2 above)
Comment (TristonianJones) [reply ext/strings_test.go]: Done.

**Pull request merged**